### ISSUE TITLE
ContextResult::kFatalFailure: too few vertex attributes supported (0, should be 8).

### ISSUE THREAD

**Issue by barcoboy** [gm-vm/openfortivpn-webview#2]
After successfully compiling the Qt version of openfortivpn-webview on my older laptop running Linux AntiX 19.5 (32 bit), when I run the executable get a black screen with the title bar containing the URL of my SAML provider, but going back to the terminal window where I launched from, I am seeing the following line scroll endlessly until I kill the process:

[9483:9518:0705/104308.408845:ERROR:context_group.cc(261)] ContextResult::kFatalFailure: too few vertex attributes supported (0, should be 8).

Any ideas how I can fix this?

**Comment by gm-vm** [comment]
That error is coming from the inner Chromium engine, the application is just a thin `QtWebEngine` wrapper.

If the SAML page loads without issues, you can try to just hide all the messages from Chromium with the command mentioned in the README, or this one:
```shell
QTWEBENGINE_CHROMIUM_FLAGS="--enable-logging --log-level=3" openfortivpn-webview
```

If that works, you could simply create an alias or a shell script to launch the application with that variable set.

**Comment by barcoboy** [comment]
I was able to get it working.  The problem is the age of my laptop (an old IBM R51), and I remember running into a similar issue when trying to get KODI working on the same machine.  Fortunately, I made notes about it, and was able to find in my notes the following:

--------

Because the on-board graphics interface of the R51 does not support OpenGL
2.0, but only version 1.4, hardware rendering cannot be done, and running
KODI will result in a black video screen.  To get KODI to run on these
machines, software rendering must be forced by running the following:

export LIBGL_ALWAYS_SOFTWARE=1

--------

So I added LIBGL_ALWAYS_SOFTWARE=1 to the startup, and now the login page is displayed and the cookie is output after authenticating.

Maybe this will help someone else out who has older hardware but still wants to use openfortivpn on it.

:-)
